src/exe/FreeC/Backend.hs

```diff
@@ -136,3 +140,21 @@ coqBackend = Backend { backendName          = "coq"
                      , backendFileExtension = "v"
                      , backendSpecialAction = createCoqProject
                      }
+
+-------------------------------------------------------------------------------
+-- Agda backend                                                              --
+-------------------------------------------------------------------------------
+
+-- | Converts an IR module to a Coq program.
+convertModuleToAgda :: IR.Module -> Application String
+convertModuleToAgda ast = do
+  ast' <- liftConverter $ Agda.Converter.convertModule ast
+  return $ showPretty $ PrettyAgda ast'
```

Comment:
This implementation can be minimized. The `PrettyAgda` isn't needed. 
```suggestion
convertModuleToAgda =
  fmap showPretty . liftConverter . Agda.Converter.convertModule
```
With this implementation the import can be simplified to `import FreeC.Backend.Agda.Pretty ()`.